WikiDrips/WikiRequest.swift

```diff
@@ -42,32 +40,50 @@ public class WikiRequest
                     print("error trying to convert data to JSON")
                     return
                 }
-                if let dictionary = response as? [String: Any] {
-                    if let query = dictionary["query"] as? [String: Any] {
-                        if let results = query["search"] as? [Any] {
-                            for item in results {
-                                guard let item = item as? [String: Any] else {
-                                    break
-                                }
-                                if let wikiDoc = WikiDoc(data: item) {
-                                    self.wikiDocs.insert([wikiDoc], at: 0)
-                                }
-                            }
-                        }
+                
+                guard let dictionary = response as? [String: Any],
+                    let query = dictionary["query"] as? [String: Any],
+                    let results = query["search"] as? [Any] else {
+                    return
+                }
+                var wikiDocs = [WikiDoc]()
+                let isoDateFormatter = ISO8601DateFormatter()
+                for item in results {
+                    guard let item = item as? [String: Any],
+                        let title = item["title"] as? String,
+                        let isoDate = item["timestamp"] as? String,
+                        let date = isoDateFormatter.date(from: isoDate) else {
+                        break
```

Comment:
With a `for` loop you probably want `continue` instead of `break` so we go to the next element. But actually, this is the perfect use case for `flatMap`:
```
let wikiDocs: [WikiDoc] = results.flatMap {
                    guard let item = $0 as? [String: Any],
                        let title = item["title"] as? String,
                        let isoDate = item["timestamp"] as? String,
                        let date = isoDateFormatter.date(from: isoDate) else {
                            return nil
                    }
                    return WikiDoc(title: title, date: date)
                }
```
Here's some fun reading on [these](https://digitalleaves.com/blog/2017/02/sequence-hacking-in-swift-i-map-flatmap-sort-filter-reduce/).